packages/opencensus-exporter-jaeger/src/jaeger.ts

```diff
@@ -0,0 +1,197 @@
+import ThriftUtils from './jaeger-driver/thrift';
+import UDPSender from './jaeger-driver/udp_sender';
+import Utils from './jaeger-driver/util';
+
+/**
+ * Options for Jaeger configuration
+ */
+export interface JaegerTraceExporterOptions extends types.ExporterConfig {
+  serviceName: string;
+  tags?: Tag[];
+  host?: string;
+  port?: number;
+  maxPacketSize?: number;
+}
+
+
+/** Format and sends span information to Jaeger */
+export class JaegerTraceExporter implements types.Exporter {
+  // Name of the tag used to report client version.
+  static JAEGER_OPENCENSUS_EXPORTER_VERSION_TAG_KEY =
+      'opencensus.exporter.jaeger.version';
+  // host name of the process.
+  static TRACER_HOSTNAME_TAG_KEY = 'opencensus.exporter.jaeger.hostname';
+  //  ip of the process.
+  static PROCESS_IP = 'ip';
+
+  private process: Process;
+  private logger: types.Logger;
+  private sender: UDPSender;
+  checkCallback?: (exporter: JaegerTraceExporter) => void;
+  spanBuffer: types.Span[] = [];
+  successBuffer: types.Span[] = [];
+  /** Timeout control */
+  /** Max time for a buffer can wait before being sent */
+  private bufferTimeout: number;
+  /** Manage when the buffer timeout needs to be reseted */
+  private resetTimeout = false;
+  /** Indicates when the buffer timeout is running */
+  private timeoutSet = false;
+
+
+  constructor(options: JaegerTraceExporterOptions) {
+    const pjson = require('../../package.json');
+    this.logger = options.logger || logger.logger('debug');
+    this.bufferTimeout = options.bufferTimeout;
+    this.sender = new UDPSender(options);
+    const tags = options.tags || [];
+    tags[JaegerTraceExporter.JAEGER_OPENCENSUS_EXPORTER_VERSION_TAG_KEY] =
+        `opencensus-exporter-jaeger-${pjson.version}`;
+    tags[JaegerTraceExporter.TRACER_HOSTNAME_TAG_KEY] = os.hostname();
+    tags[JaegerTraceExporter.PROCESS_IP] = Utils.ipToInt(Utils.myIp());
+
+    this.process = {
+      serviceName: options.serviceName,
+      tags: options.tags ? ThriftUtils.getThriftTags(tags) : [],
+    };
+    this.sender.setProcess(this.process);
+  }
+
+  /**
+   * Is called whenever a span is ended.
+   * @param root the ended span
+   */
+  onEndSpan(root: types.RootSpan) {
+    this.logger.debug('onEndSpan: adding rootSpan: %s', root.name);
+    this.addSpanToSenderBuffer(root)
+        .then(result => {
+          this.addToBuffer(root, result as number);
+          for (const span of root.spans) {
+            this.addSpanToSenderBuffer(span)
+                .then(result => {
+                  this.addToBuffer(span, result as number);
+                })
+                .catch(err => {
+                  return;
+                });
+          }
+        })
+        .catch(err => {
+          return;
+        });
+
+    // Set a buffer timeout
+    this.setBufferTimeout();
+  }
+
+  /** Not used for this exporter */
+  onStartSpan(root: types.RootSpan) {}
+
+  private addToBuffer(span: types.Span, numSpans: number) {
```

Comment:
Can you document what this method is doing? I'm not sure why logic branches based on whether `numSpans > 0`.